lib/elixir/lib/module/types/infer.ex

```diff
@@ -485,9 +548,13 @@ defmodule Module.Types.Infer do
 
   def subtype?(left, right, _context), do: left == right
 
+  @doc """
+  Returns a "simplified" union using `subtype?/3` to remove redundant
+  types. Due to limitations in `subtype?/3` some overlapping types
```

Comment:
Which limitations? Let’s document it explicitly and what is the impact of them.

Reply:
The limitations are expanded on in the `subtype?/3` docs.